lib/cli/run.js

```diff
@@ -0,0 +1,403 @@
+  runShowReporters,
+  handleExtensions,
+  handleFiles,
+  handleRequires,
+  validatePlugin,
+  runMocha
+} = require('./run-helpers');
+
+/**
+ * Logical option groups
+ */
+const GROUPS = {
+  FILES: 'File Handling',
+  FILTERS: 'Test Filters',
+  NODEJS: 'Node.js & V8',
+  OUTPUT: 'Reporting & Output',
+  RULES: 'Rules & Behavior',
+  CONFIG: 'Config Files'
+};
+
+/**
+ * Placeholder value to force config parsing function to run.
+ */
+const CONFIG_PATH_PLACEHOLDER = '__MOCHA_CONFIG_PATH__';
+
+exports.command = '$0 [spec..]';
+
+exports.describe = 'Run tests with Mocha';
+
+exports.builder = yargs =>
+  yargs
+    .showHidden(
+      'show-node-options',
+      'Show supported Node.js & V8 options; use with "--help"'
+    )
+    .options({
+      'allow-uncaught': {
+        description: 'Allow uncaught errors to propagate',
+        group: GROUPS.RULES,
+        type: 'boolean'
+      },
+      'async-only': {
+        alias: 'A',
+        description:
+          'Require all tests to use a callback (async) or return a Promise',
+        group: GROUPS.RULES,
+        type: 'boolean'
+      },
+      'check-leaks': {
+        description: 'Check for global variable leaks',
+        group: GROUPS.RULES,
+        type: 'boolean'
+      },
+      'expose-gc': {
+        alias: 'gc',
+        description: 'Equivalent to node flag "--expose-gc"',
+        hidden: true,
+        type: 'boolean'
+      },
+      'forbid-only': {
+        description: 'Fail if exclusive test(s) encountered',
+        group: GROUPS.RULES,
+        type: 'boolean'
+      },
+      'forbid-pending': {
+        description: 'Fail if pending test(s) encountered',
+        group: GROUPS.RULES,
+        type: 'boolean'
+      },
+      'full-trace': {
+        description: 'Display full stack traces',
+        group: GROUPS.OUTPUT,
+        type: 'boolean'
+      },
+      'inline-diffs': {
+        description:
+          'Display actual/expected differences inline within each string',
+        group: GROUPS.OUTPUT,
+        type: 'boolean'
+      },
+      'no-colors': {
+        alias: 'C',
+        conflicts: ['colors'],
+        description: 'Disable color output',
+        group: GROUPS.OUTPUT,
+        type: 'boolean'
+      },
+      'no-timeouts': {
+        conflicts: ['timeout'],
+        description: 'Disable timeouts',
+        group: GROUPS.RULES,
+        type: 'boolean'
+      },
+      'reporter-options': {
+        alias: 'O',
+        coerce: (opts = '') =>
+          opts.split(',').reduce((acc, opt) => {
+            const pair = opt.split('=');
+
+            if (pair.length > 2 || !pair.length) {
+              throw new Error(`invalid reporter option '${opt}'`);
+            }
+
+            acc[pair[0]] = pair.length === 2 ? pair[1] : true;
+            return acc;
+          }, {}),
+        description: 'Reporter-specific options (<k=v,[k1=v1,..]>)',
+        group: GROUPS.OUTPUT,
+        requiresArg: true,
+        type: 'string'
+      },
```

Comment:
For what it's worth, this needs to be replaced with `--reporter-option` that can be used multiple times (collected). Current version doesn't allow for passing arguments with commas in the value. Given druthers, the routine would allow the value to be hierarchially-capable (#3487), beginning with the affected reporter's name. Would also simplify "commenting out" a single reporter option without having to "edit" the existing.

Example:
```bash
$ mocha --reporter xunit --reporter-option xunit.outfile="xunit.xml" --reporter-option xunit.suiteName="Surely Thou Jest" 
```
equivalent to
```bash
$ mocha --reporter xunit --reporter-options outfile="xunit.xml",suiteName="Surely Thou Jest" 
```

Reply:
Agree. This is pretty trivial since yargs does a bunch of this for us.  Let's soft-deprecate `--reporter-options`. Sound good?

- [ ] implement `--reporter-option`; soft-deprecate `--reporter-options`

We could go a step further and soft-deprecate anything else that wants a comma-separated parameter..

Reply:
I'm not going to do this right now.  It should get done though.

Reply:
https://github.com/mochajs/mocha/issues/3564#issuecomment-458740928